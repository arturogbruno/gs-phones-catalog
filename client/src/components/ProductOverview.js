import React from "react";
import { Link } from "react-router-dom";

const ProductOverview = ({ category, product }) => {
    const { id, manufacturer, name, price, imageFileName } = product;

    return (
        <div className="ProductOverview">
            <Link to={`/${category}/${id}`}>
                <img src={`images/${imageFileName}`} alt={`${manufacturer} ${name}`} />
                <h3>{manufacturer}</h3>
                <h3>{name}</h3>
                <h3>{price} â‚¬</h3>
            </Link>
        </div>
    )
}

export default ProductOverview;