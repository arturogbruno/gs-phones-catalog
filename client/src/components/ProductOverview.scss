.ProductOverview {
    width: 100%;
    background-color: rgba(255, 255, 255, 0.4);
    margin-bottom: 50px;

    a {
        display: flex;
        justify-content: space-between;
        align-items: center;

        img {
            width: 10%; 
        }
    }
}