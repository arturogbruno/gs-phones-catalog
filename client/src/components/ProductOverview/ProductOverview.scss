@import "../../styles/mixins.scss";

.ProductOverview {
    width: 100%;
    background: linear-gradient(90deg, rgba(255,255,255,1) 15%, rgba(255,255,255,0.5) 70%, rgba(255,255,255,0.5) 100%);
    margin-bottom: 50px;

    a {
        height: 100px;
        display: flex;
        justify-content: space-between;
        align-items: center;
        text-decoration: none;
        color: rgb(35, 35, 35);
        transition: all 0.2s ease-out;

        &:hover {
            box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
        }

        img {
            height: 100%;
        }

        .productMainInfo {
            width: 60%;
            display: flex;
            justify-content: space-between;
            align-items: center;
            margin: 0;
            padding-right: 10px;
            font-size: 0.7em;

            .makerModel {
                display: flex;
                flex-direction: column;

                h2 {
                    line-height: 2rem;
                }
            }

            .price {
                font-style: italic;
            }

            @include md {
                font-size: 1em;
                margin: 0 10%;
                padding-right: 0;
            }
        }

        @include md {
            height: 150px;
        }
    }
}