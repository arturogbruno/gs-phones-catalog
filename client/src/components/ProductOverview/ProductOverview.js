import React from "react";
import { Link } from "react-router-dom";
import "./ProductOverview.scss";

const ProductOverview = ({ category, product }) => {
    const { id, manufacturer, name, price, imageFileName } = product;

    return (
        <div className="ProductOverview">
            <Link to={`/${category}/${id}`}>
                <img src={`images/${imageFileName}`} alt={`${manufacturer} ${name}`} />
                <div className="productMainInfo">
                    <div className="makerModel">
                        <h2>{manufacturer}</h2>
                        <h2>{name}</h2>
                    </div>
                    <h2 className="price">{price} â‚¬</h2>
                </div>
            </Link>
        </div>
    )
}

export default ProductOverview;