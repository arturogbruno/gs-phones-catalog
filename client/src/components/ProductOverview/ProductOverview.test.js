import React from "react";
import { shallow } from "enzyme";
import ProductOverview from "./ProductOverview";

describe("testing ProductOverview component", () => {
    it("should render ProductOverview component", () => {
        const mockCategory = "phones";
        const mockProduct = {
            "id": 2,
            "name": "Honor 10",
            "manufacturer": "Huawei",
            "description": "Great phone with an excellent interface. One of the best mid price range phones in the market.",
            "color": "blue",
            "price": 399,
            "imageFileName": "Honor_10.png",
            "screen": "5,84 inch Full-HD",
            "processor": "Kirin 970",
            "ram": 6
        };

        expect(shallow(<ProductOverview category={mockCategory} product={mockProduct} />)).toMatchSnapshot();
    });
});