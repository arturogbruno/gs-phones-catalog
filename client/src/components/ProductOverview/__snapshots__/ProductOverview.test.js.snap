// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ProductOverview component should render ProductOverview component 1`] = `
<div
  className="ProductOverview"
>
  <Link
    to="/phones/2"
  >
    <img
      alt="Huawei Honor 10"
      src="images/Honor_10.png"
    />
    <div
      className="productMainInfo"
    >
      <div
        className="makerModel"
      >
        <h2>
          Huawei
        </h2>
        <h2>
          Honor 10
        </h2>
      </div>
      <h2
        className="price"
      >
        399
         â‚¬
      </h2>
    </div>
  </Link>
</div>
`;
